hazelcast/src/main/java/com/hazelcast/map/impl/EntryViews.java

```diff
@@ -52,17 +53,33 @@ public static <K, V> EntryView<K, V> createSimpleEntryView(K key, V value, Recor
 
     public static <K, V> WanMapEntryView<K, V> createWanEntryView(Data key, Data value,
                                                                   Record<V> record, ExpiryMetadata expiryMetadata,
-                                                                  SerializationService serializationService) {
+                                                                  SerializationService serializationService,
+                                                                  boolean isPerEntryStatsEnabled) {
         return new WanMapEntryView<K, V>(key, value, serializationService)
                 .withCost(record.getCost())
                 .withVersion(record.getVersion())
                 .withHits(record.getHits())
                 .withLastAccessTime(record.getLastAccessTime())
-                .withLastUpdateTime(record.getLastUpdateTime())
+                .withLastUpdateTime(calculateLastUpdateTime(isPerEntryStatsEnabled, record, expiryMetadata))
                 .withCreationTime(record.getCreationTime())
                 .withLastStoredTime(record.getLastStoredTime())
                 .withTtl(expiryMetadata.getTtl())
                 .withMaxIdle(expiryMetadata.getMaxIdle())
                 .withExpirationTime(expiryMetadata.getExpirationTime());
     }
+
+    private static <V> long calculateLastUpdateTime(boolean isPerEntryStatsEnabled,
+                                                    Record<V> record,
+                                                    ExpiryMetadata expiryMetadata) {
+        if (isPerEntryStatsEnabled) {
+            return record.getLastUpdateTime();
+        }
+
+        if (expiryMetadata != ExpiryMetadata.NULL) {
+            return expiryMetadata.getLastUpdateTime();
+        }
+
+        // Record.UNSET
+        return -1;
```

Comment:
Maybe it will be better to use the `Record.UNSET` everywhere.

Reply:
I would need to make that public. Would you prefer that?

Reply:
nvm, it was an interface, i was blind

Reply:
https://github.com/hazelcast/hazelcast/pull/23279/commits/375089114de8adceed51b08918be5fad48f3ce9a